### PR TITLE
Feature/bma/notification decryption

### LINKED ISSUE #1178: Some notifications don't decrypt
### Steps to reproduce

1. Rolling down the notification drawer of Android
2. Looking at EX notifications

#### What did you expect?
See a line of text with the notification

#### What happened instead?
Some notifications only contain the line "message"

Screenshot_20230829-190635

### Your phone model

Pixel 5

### Operating system version

Android 13

### Application version and app store

0.1.6-nightly

### Homeserver

element.io

### Will you send logs?

No

### Are you willing to provide a PR?

No

### PR DESCRIPTION
**Pull request by bmarty**
## Type of change

- [ ] Feature
- [x] Bugfix
- [ ] Technical
- [ ] Other :

## Content

Ensure that the sync service is started when handling a Push.

Add to implement a `reason`, to ensure that we call `syncService.stop()` only when no code is using it.

## Motivation and context

Fixes #1178 

## Screenshots / GIFs

NA

## Tests

From https://github.com/vector-im/element-x-android/issues/1178#issuecomment-1702485220:

- Alice and Bob in room. Bob is using EAX
- bob kills EAX
- Alice rotates their megolm keys by typing '/discardsession' in this room
- Alice sends a message
- Bob should see the notification decrypted on their android device

With this change, Bob can see the message in the notification.

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s): 13

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-x-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR

### CODE REVIEW

**Comment by jmartinesp** [comment]
We display notifications too when the app is in foreground for other rooms than the one the user is in. Has this been taken into account in the Rust SDK?

**Comment by jmartinesp** [comment]
It seems like the sample project needs some fixes.

File: libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/sync/RustSyncService.kt [hunk 1]
```diff
@@ -33,19 +36,36 @@ class RustSyncService(
     private val innerSyncService: SyncServiceInterface,
     sessionCoroutineScope: CoroutineScope
 ) : SyncService {
+    private val mutex = Mutex()
+    private val startSyncReasonSet = mutableSetOf<StartSyncReason>()
```
Comment (bnjbvr) [inline libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/sync/RustSyncService.kt:40]: I don't know enough about Kotlin, but can the mutex hold the startSyncReasonSet itself, like we'd do in Rust `Mutex<HashSet<StartSyncReason>>`? That would make it more obvious that one needs the mutex to read/write to the set.

**Review by bnjbvr — APPROVED** [review]
Makes sense :+1:

File: libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/sync/RustSyncService.kt (hunk 1 above)
Comment (jmartinesp) [reply libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/sync/RustSyncService.kt:40]: I don't think we can do something like that, `Mutex` in Kotlin can't hold values.

**Review by jmartinesp — APPROVED** [review]
Assuming that 'killing the app' means just removing it from recents (which should keep the notification receiver alive) it seems to be working fine. Once we fix the sample project issue it should be ok to merge.

**Pull request merged**